src/data_snack/snack.py

```diff
@@ -163,4 +150,4 @@ def keys(self, cls: Type[Entity]) -> List[Text]:
         :param cls: Entity type
         :return: a list of keys
         """
-        return self.connection.keys(cls, pattern=self.key_factory.get_pattern(cls.__name__))
+        return self.connection.keys(pattern=self.key_factory(cls, []).get_pattern(cls.__name__))
```

Comment:
Does it work? `I think the pattern should have `-*` suffix

Reply:
You're right, fixed. I also added a test for it (it was missed, that is why I missed the case too, since only Mongo was tested).